packages/grafana-ui/src/components/Collapse/Collapse.tsx

```diff
@@ -142,15 +142,25 @@ export const Collapse = ({
   const panelClass = cx([style.collapse, className]);
   const loaderClass = loading ? cx([style.loader, style.loaderActive]) : cx([style.loader]);
   const headerClass = collapsible ? cx([style.header]) : cx([style.headerCollapsed]);
+  const idContent = label?.toString();
 
   return (
     <div className={panelClass}>
       <div className={headerClass} onClick={onClickToggle}>
-        {collapsible && <Icon className={style.icon} name={isOpen ? 'angle-down' : 'angle-right'} />}
-        <div className={cx([style.headerLabel])}>{label}</div>
+        {collapsible && (
+          <IconButton
+            aria-expanded={isOpen}
+            aria-controls={idContent}
+            className={cx([style.icon])}
+            name={isOpen ? 'angle-down' : 'angle-right'}
+          />
+        )}
+        <div className={cx([style.headerLabel])} id={`header-${idContent}`}>
```

Comment:
Not sure if this `id` (used in the section div to set `aria-labelledby`) should go here or in the wider div, the one that has the `onClick` ?

Reply:
`label` is a `ReactNode`, so you don't want to call `.toString()` on it:

![image](https://user-images.githubusercontent.com/46142/196470055-a60ff69b-1605-4320-a1b1-c96ad82dea75.png)

Also, IDs must be unique, but you have no guarantee that someone else won't be using a Collapse with title="Advanced" on the same page (especially when you're a data source query editor being rendered on Explorer, or in panel edit with other query editors!) 

Instead, you can `import { useId } from '@react-aria/utils';` and use that to generate an ID that'll be both unique and stable.